### DISCUSSION TITLE
error message "Operator::MultTranspose() is not overloaded!"

### DISCUSSION THREAD

**Discussion by markusrenoldner** [mfem/mfem#3475] [Q&A] [answered]
Hello community,

I am trying to solve the system $Ax = b$ after making it symmetric by applying the transposed of $A$ from both sides:

$$A^TAx=A^Tb $$

The following (simplified) code results in the error message from the discussion title:
```
mfem::BlockOperator A (offsets);
mfem::Operator *A_constrained;
mfem::Vector B, X;
A.FormLinearSystem(essdofs, x, b, A_constrained, X, B);

mfem::TransposeOperator AT (A_constrained);
mfem::ProductOperator ATA (&AT, A_constrained, false, false);
mfem::Vector ATB;
A.MultTranspose(B,ATB);
mfem::MINRES(*A_constrained, ATB, X, 0, iter, tol, tol); 
```

The same error message appears if I replace `A.MultTranspose(B,ATB)` with `AT.Mult(B,ATB)`.
I found what that the `operator.hpp` file says about the `mfem::MultTranspose()` method:
> Action of the transpose operator: `y=A^t(x)`. The default behavior in class Operator is to generate an error.

Does that mean, the action of the transposed operator is supposed to raise an error? How do you apply the transposed operator then?

(i got asked to tag specific people from the community when asking questions @dylan-copeland)

**Comment by pazner** [comment, marked as answer]
Not every `Operator` implements `MultTranspose`. For example, `ConstrainedOperator` (which is what is used with matrix-free/partial assembly) does not implement `MultTranspose`. I believe this should be relatively easy to implement though. So it depends on what type of `Operator` you have — sparse matrices like `SparseMatrix` and `HypreParMatrix` both implement `MultTranspose`.

**Comment by markusrenoldner** [comment]
Thank you for your quick answer!
